crates/engine/tree/src/tree/mod.rs

```diff
@@ -500,24 +500,23 @@ where
     /// Run the engine API handler.
     ///
     /// This will block the current thread and process incoming messages.
-    pub fn run(mut self) {
+    pub fn run(mut self) -> Result<(), InsertBlockFatalError> {
```

Comment:
in this method I would prefer that the signature remain as `run(mut self)`, and instead we handle the `on_engine_message` like this:
```rust
if let Err(err) = self.on_engine_message(msg) {
    // some error! message
    return
}
```

Reply:
Okay no problem, I was not sure about this detail. Thank you for clarifying.